source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -144,27 +109,21 @@ namespace aspect
     double
     AdiabaticBoundary<dim>::initial_temperature (const Point<dim> &) const
     {
-      Assert (false, ExcNotImplemented());
+      AssertThrow (false, ExcMessage ("The 'adiabatic boundary' initial temperature plugin is only implemented for 3d cases."));
```

Comment:
This function is a bit confusing, because it is not a template specialization (I guess it is only called in 2d, because you provide a specialization for 3d below). Can you change it to the same format as the `AdiabaticBoundary<2>::get_isotherm_depth()` function above? I.e. please make this function a template specialization for dim==2.